tests/mmm/test_budget_optimizer.py

```diff
@@ -630,3 +630,275 @@ def test_mmm_optimize_budget_callback_parametrized(dummy_df, dummy_idata, callba
     assert hasattr(opt_result, "success")
     assert hasattr(opt_result, "x")
     assert hasattr(opt_result, "fun")
+
+
+@pytest.mark.parametrize(
+    "budget_distribution_over_period, num_periods, should_error, error_message",
+    [
+        # Valid case: uniform distribution
+        (
+            {
+                "channel_1": [0.25, 0.25, 0.25, 0.25],
```

Comment:
Slightly different angle is to just test that the optimizer arrives at the same result when you put in a uniform flight and when you don't pass a flight.